creator-node/src/RehydrateIpfsQueue.js

```diff
@@ -0,0 +1,102 @@
+const Bull = require('bull')
+const config = require('./config')
+const { rehydrateIpfsFromFsIfNecessary, rehydrateIpfsDirFromFsIfNecessary } = require('./utils')
+const { logger: genericLogger } = require('./logging')
+
+const PROCESS_NAMES = Object.freeze({
+  rehydrate_dir: 'rehydrate_dir',
+  rehydrate_file: 'rehydrate_file'
+})
+
+class RehydrateIpfsQueue {
+  constructor () {
+    this.queue = new Bull(
+      'rehydrateIpfs',
+      {
+        redis: {
+          host: config.get('redisHost'),
+          port: config.get('redisPort')
+        }
+      }
+    )
+
+    this.queue.process(PROCESS_NAMES.rehydrate_file, config.get('rehydrateMaxConcurrency'), async (job, done) => {
+      const { multihash, storagePath, filename, logContext } = job.data
+
+      this.logStatus(logContext, `Processing a rehydrateIpfsFromFsIfNecessary task for ${multihash}`)
+      // TODO: all the expected rehydrate errors are caught/logged. so catch might not actually catch error
+      // can instead of logging errors in rehydrate code, throw errors and handle here
```

Comment:
is this comment out of date? because errors are not thrown, just caught and handled here it seems

Reply:
yeah, all the expected errors in the rehydrate methods are caught with the many try/catch blocks. this try/catch block here was a safety net as to if something unexpected were to fail. 

tbh I dont think this try/catch is really necessary? what do you think

Reply:
i think the way the code here is structured is exactly right - we don't want uncaught errors since it will kill the queue itself. however the TODO can be updated to reflect that